substrate/frame/contracts/fixtures/build.rs

```diff
@@ -0,0 +1,277 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Compile contracts to wasm and RISC-V binaries.
+use anyhow::Result;
+use parity_wasm::elements::{deserialize_file, serialize_to_file, Internal};
+use std::{
+	env, fs,
+	hash::Hasher,
+	path::{Path, PathBuf},
+	process::Command,
+};
+use twox_hash::XxHash32;
+
+/// Read the file at `path` and return its hash as a hex string.
+fn file_hash(path: &Path) -> String {
+	let data = fs::read(path).expect("file exists; qed");
```

Comment:
Total nit, this isn't actually true but fine for me if it is enforced further up (also it's a build script and not a critical path in the runtime so not the end of the world if we panic here I guess)